### ISSUE TITLE
integration_test package pauses isolates and calling pumpAndSettle() never finishes.

### ISSUE THREAD

**Issue by drewjbartlett** [flutter/flutter#73355]
## Steps to Reproduce

When trying to write tests with the new `integration_test` package it firstly will hang because it pauses isolates and never resumes them. I can confirm this because an http request is made and a json transformer runs in an isolate. Therefore, the response never finishes. When I remove the transformer, the response comes back from the request. 

1. `flutter drive  --driver=test_driver/integration_driver.dart  --target=integration_test/app_test.dart`

Test file:

```dart
void main() {
  IntegrationTestWidgetsFlutterBinding.ensureInitialized();

  testWidgets('it shows the home tab', (
    WidgetTester tester,
  ) async {
    await app.main();

    /// Even when disabling the json parsing isolate, this never completes.
    await tester.pumpAndSettle();

    expect(find.byKey(const Key('tab-home')), findsOneWidget);
  });
}
```

When `pumpAndSettle()` runs, the tests hangs and only occasionally says:

```
VMServiceFlutterDriver: request_data message is taking a long time to complete...
```

When running `pump()` instead I'm able to trigger tap events and the app does not hang but none of my assertions will return true. 

## Logs

<details>
<summary>Logs</summary>

```
flutter: Observatory listening on http://127.0.0.1:60529/aAM9-23tIj0=/
VMServiceFlutterDriver: Connecting to Flutter application at http://127.0.0.1:60529/aAM9-23tIj0=/
VMServiceFlutterDriver: Isolate found with number: 2871742328872651
VMServiceFlutterDriver: Isolate is paused at start.
VMServiceFlutterDriver: Attempting to resume isolate
flutter: 00:00 +0: it shows the home tab
VMServiceFlutterDriver: Connected to Flutter application.
flutter: *** Request ***
flutter: uri: http://localhost:9999/api/auth/user
flutter: method: POST
flutter: responseType: ResponseType.json
flutter: followRedirects: true
flutter: connectTimeout: 60000
flutter: receiveTimeout: 60000
flutter: extra: {}
flutter: headers:
flutter:  accept: application/json
flutter:  content-type: application/json; charset=utf-8
flutter:
```
(hangs here because the isolate is parsing json and is paused).

Flutter Doctor: 

```
[✓] Flutter (Channel stable, 1.22.5, on macOS 11.1 20C69 darwin-x64, locale en-US)
    • Flutter version 1.22.5 at /Users/drewbartlett/flutter
    • Framework revision 7891006299 (4 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4
```
</details>

To sum this up:

1. Isolates are paused. How can we resume them with `integration_test`?
2. Any idea why `pumpAndSettle()` hangs?

I've been stuck on this for almost 2 days now and would appreciate any feedback! Thank you!

**Comment by MostafaSolimanMO** [comment]
Same problem, any solutions or workarounds?

**Comment by darshankawar** [comment]
@drewjbartlett 
I tried the official `integration_test` plugin [example](https://pub.dev/packages/integration_test) and modified it's `example_test.dart` to use your code sample which worked properly.

```
void main() {
    IntegrationTestWidgetsFlutterBinding.ensureInitialized();
    group('Counter', () {
      testWidgets("failing test example", (WidgetTester tester) async {
        await app.main();
        await tester.pumpAndSettle();
        expect(2 + 2, equals(4));
      //  throw Exception();
      });
    });
  }
```

```
I/flutter (17932): 00:00 +0: Counter failing test example
VMServiceFlutterDriver: Connected to Flutter application.
I/flutter (17932): 00:01 +1: (tearDownAll)
I/flutter (17932): 00:01 +2: All tests passed!
All tests passed.
Stopping application instance.
dhs@Dhss-MacBook-Pro example % 

```

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-IN)
    • Flutter version 1.22.5 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 7891006299 (4 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android
      6.0 (API 23) (emulator)

! Doctor found issues in 1 category.



```

</details>

Can you try the same and compare if it still hangs for you ?

**Comment by drewjbartlett** [comment]
If I try this with my app, here are the results: 

Test:

```dart
void main() {
  IntegrationTestWidgetsFlutterBinding.ensureInitialized();

  testWidgets('it shows all the tabs', (
    WidgetTester tester,
  ) async {
    await app.main();

    await tester.pumpAndSettle();

    expect(2 + 2, equals(4));
  });
}
```

Result:

```bash
flutter: Observatory listening on http://127.0.0.1:52257/T8wfZR260kc=/
VMServiceFlutterDriver: Connecting to Flutter application at http://127.0.0.1:52257/T8wfZR260kc=/
VMServiceFlutterDriver: Isolate found with number: 369897860022107
VMServiceFlutterDriver: Isolate is paused at start.
VMServiceFlutterDriver: Attempting to resume isolate
flutter: 00:00 +0: it shows all the tabs
VMServiceFlutterDriver: Connected to Flutter application.
flutter: *** Request ***
flutter: uri: http://127.0.0.1:9999/api/auth/user
flutter: method: POST
flutter: responseType: ResponseType.json
flutter: followRedirects: true
flutter: connectTimeout: 60000
flutter: receiveTimeout: 60000
flutter: extra: {}
flutter: headers:
flutter:  accept: application/json
flutter:  content-type: application/json; charset=utf-8
flutter:
VMServiceFlutterDriver: request_data message is taking a long time to complete...
```

It continues to hang. If I remove the isolate for transforming JSON and run again, the request completes but the `expect` won't run and I still get 

`VMServiceFlutterDriver: request_data message is taking a long time to complete...`

**Comment by darshankawar** [comment]
@drewjbartlett 
Can you provide the entire code sample (minimal reproducible) including `main.dart` that we can use and verify this issue ?
Thanks.

**Comment by drewjbartlett** [comment]
@darshankawar here's my `main.dart` file with some comments to explain what's going on. Is this enough to help?


```dart
void main() async {
  WidgetsFlutterBinding.ensureInitialized();
 
  /// load DotEnv() files
  await initializeEnvironment();

  /// initialize IOC Container
  IOC.init();

  /// Register event listeners
  registerEventListeners();

  EventBus eventBus = IOC.container<EventBus>();

  eventBus.fire(AppLaunchedEvent());

  /// run the app with providers wrapped around it.
  runApp(
    MultiProvider(
      providers: [
        ChangeNotifierProvider.value(
          value: IOC.container<AuthState>(),
        ),
        ChangeNotifierProvider.value(
          value: IOC.container<TabNavigatorState>(),
        ),
      ],
      child: App(eventBus: eventBus),
    ),
  );
}
```

**Comment by darshankawar** [comment]
<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.26.0-2.0.pre.275, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-IN)
    • Flutter version 1.26.0-2.0.pre.275 at
      /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision d9044a8a61 (2 days ago), 2021-01-09 04:04:04 -0500
    • Engine revision caf6a8191f
    • Dart version 2.12.0 (build 2.12.0-204.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (4 available)
    • sdk gphone x86 64 arm64 (mobile)    • emulator-5554
      • android-x64    • Android 11 (API 30) (emulator)
    • iPhone SE (2nd generation) (mobile) • 6C85835D-FBFD-4AB3-8DE8-B4FAD35E5367
      • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    • macOS (desktop)                     • macos
      • darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)                        • chrome
      • web-javascript • Google Chrome 87.0.4280.141

! Doctor found issues in 1 category.

```

</details>

**Comment by lukaruna** [comment]
I was able to work around a similar issue by modifying the `integrationDriver` at integration_test.dart. Does it work for you?

```

import 'dart:async';
import 'dart:io';

import 'package:flutter_driver/flutter_driver.dart';
import 'package:integration_test/common.dart';
import 'package:integration_test/integration_test_driver.dart';

Future<void> main() => integrationDriver();

Future<void> integrationDriver({
  Duration timeout = const Duration(minutes: 1),
  ResponseDataCallback responseDataCallback = writeResponseData,
}) async {
  final driver = await FlutterDriver.connect();

  final workaround = IsolatesWorkaround(driver);
  await workaround.resumeIsolates();

  final jsonResult = await driver.requestData(null, timeout: timeout);
  final response = Response.fromJson(jsonResult);
  await driver.close();

  if (response.allTestsPassed) {
    print('All tests passed.');
    if (responseDataCallback != null) {
      await responseDataCallback(response.data);
    }
    exit(0);
  } else {
    print('Failure Details:\n${response.formattedFailureDetails}');
    exit(1);
  }
}

class IsolatesWorkaround {
  IsolatesWorkaround(this._driver, {this.log = false});

  final FlutterDriver _driver;
  final bool log;
  // ignore: cancel_subscriptions
  StreamSubscription _streamSubscription;

  /// workaround for isolates
  /// https://github.com/flutter/flutter/issues/24703
  Future<void> resumeIsolates() async {
    final vm = await _driver.serviceClient.getVM();
    // // unpause any paused isolated
    for (final isolateRef in vm.isolates) {
      final isolate = await isolateRef.load();
      if (isolate.isPaused) {
        await isolate.resume();
        if (log) {
          print('Resuming isolate: ${isolate.numberAsString}:${isolate.name}');
        }
      }
    }
    if (_streamSubscription != null) {
      return;
    }
    _streamSubscription = _driver.serviceClient.onIsolateRunnable
        .asBroadcastStream()
        .listen((isolateRef) async {
      final isolate = await isolateRef.load();
      if (isolate.isPaused) {
        await isolate.resume();
        if (log) {
          print('Resuming isolate: ${isolate.numberAsString}:${isolate.name}');
        }
      }
    });
  }
}
```

**Comment by samuelstroschein** [comment]
Workaround:

```dart
Future<void> pumpForSeconds(WidgetTester tester, int seconds) async {
  bool timerDone = false;
  Timer(Duration(seconds: seconds), () => timerDone = true);
  while (timerDone != true) {
    await tester.pump();
  }
}
```

**Comment by amorenew** [comment]
@darshankawar what is the alternative to isolateRef.load in Flutter 2 as load is not supported
The whole workaround is not supported in Flutter 2

**Comment by lmordell** [comment]
@drewjbartlett , do you have any infinite animations in your app? `pumpAndSettle` won't work if there's an infinite animation like a `CircularProgressIndicator`. Try removing the animation and trying again. If that works, use `tester.pump(Duration(seconds: N))`, testing different values for N.

**Comment by adityadroid** [comment]
> @drewjbartlett , do you have any infinite animations in your app? `pumpAndSettle` won't work if there's an infinite animation like a `CircularProgressIndicator`. Try removing the animation and trying again. If that works, use `tester.pump(Duration(seconds: N))`, testing different values for N.

This solution works. Can you give an explanation why pumpAndSettle does not work with CircularProgressIndicator & `CupertinoActivityIndicator`?

**Comment by lmordell** [comment]
@adityadroid, sure. Take a look at [the documentation for pumpAndSettle](https://api.flutter.dev/flutter/flutter_test/WidgetTester/pumpAndSettle.html).

> Repeatedly calls pump with the given duration until there are no longer any frames scheduled. This will call pump at least once, even if no frames are scheduled when the function is called, to flush any pending microtasks which may themselves schedule a frame.

> This essentially waits for all animations to have completed.

> If it takes longer that the given timeout to settle, then the test will fail (this method will throw an exception). **_In particular, this means that if there is an infinite animation in progress (for example, if there is an indeterminate progress indicator spinning), this method will throw._**

**Comment by furaiev** [comment]
Maybe it helps someone.
As I have the same issue and loaders, I've modified @samuelstroschein code to:
```
Future<void> pumpTillAnimationEnd(WidgetTester tester, Duration timeout) async {
  var timerDone = false;
  Timer(timeout, () => timerDone = true);
  while (timerDone != true) {
    if (!tester.hasRunningAnimations) {
      timerDone = true;
    }
    await tester.pumpAndSettle();
  }
}
```
It finishes pumping right after the animation ends or by timeout.

**Comment by davidmartos96** [comment]
I've run into the same issue and I've created the following snippet based on the others shared in the thread.
My use case it's an infinite animation until everything is ready to show,  so I cannot use `pumpAndSettle`. The snippet works similar to the `waitFor` method from `flutter_driver` by receiving a `Finder`.
I don't know if this is the best approach, I've just started experimenting with the `integration_test`  package.

```dart
await pumpUntiFound(tester, find.byKey(ValueKey("MainScreen")));

Future<void> pumpUntiFound(
  WidgetTester tester,
  Finder finder, {
  Duration timeout = const Duration(seconds: 30),
}) async {
  bool timerDone = false;
  final timer = Timer(timeout, () => throw TimeoutException("Pump until has timed out"));
  while (timerDone != true) {
    await tester.pump();

    final found = tester.any(finder);
    if (found) {
      timerDone = true;
    }
  }
  timer.cancel();
}

```

**Comment by neiljaywarner** [comment]
@darshankawar  this would be an extremely common scenario i think; can you please consider official support?
thanks

**Comment by neiljaywarner** [comment]
@Hixie  or @RedBrogdon  any thoughts?

**Comment by neiljaywarner** [comment]
@lukaruna  this didn't "just work" with null safety, and perhaps more to the point it won't help except with flutter driver, eg won't help with firebase test lab or device farm right? eg this file optional and not run in those cases..

**Comment by tudor07** [comment]
Why doesn't flutter provide a method like @davidmartos96 `pumpUntilFound`? It seems so necessary

**Comment by matuella** [comment]
Was wondering if there were any alternatives to `FlutterDriver.waitFor` and this [snippet](https://github.com/flutter/flutter/issues/73355#issuecomment-805736745) seems to do the trick.

I could be wrong here, but I think this is a relevant missing feature from `integration_test`.

**Comment by jangruenwaldt** [comment]
This is so weird, I only get this issue on Android in my CI, Pixel 4 XL, API 29:

The same tests run reliably without this ever happening on iOS. ☹️ 
```
VMServiceFlutterDriver: Connecting to Flutter application at http://127.0.0.1:50323/...
VMServiceFlutterDriver: Isolate found with number: 4497550715463499
VMServiceFlutterDriver: Isolate is paused at start.
VMServiceFlutterDriver: Attempting to resume isolate
I/flutter ( 3417): 00:00 +0: Setup tests:  (setUpAll)
VMServiceFlutterDriver: Connected to Flutter application.
VMServiceFlutterDriver: request_data message is taking a long time to complete...
```

**Comment by lhimo** [comment]
With Flutter 2.5 missed taps and failing expect statements are behaving correctly (the test run keeps going and the current test is marked as failed). Instead, if the pumpUntilFound() function times out, the thrown exception makes the test run stop and all the subsequent test fail.

@darshankawar @davidmartos96 Do you know if there is a way to avoid this? I tried replacing the throw command with a true statement, but this way the test run is just freezing.

**Comment by lhimo** [comment]
Trying with this as of now:
```
  // pump rendering widgets until an element is found (up to 20 seconds)
  Future<void> pumpUntilFound(
    Finder finder, {
    int iterations = 200,
  }) async {
    for (int i = 1; i <= iterations; i++) {
      final found = any(finder);
      if (!found) {
        await pump(const Duration(milliseconds: 100));
      } else {
        break;
      }
    }
  }
```

**Comment by pparlova** [comment]
I've been trying with pumpUntilFound and waitFor, but still the test times out before finding the element.

**Comment by krjw-eyev** [comment]
My integration tests run on macOS but on Windows they are just stuck. `VMServiceFlutterDriver: request_data message is taking a long time to complete...` is what it says on both machines.

This happens on 2.8 whereas it would run through on both machines on 2.5.3

**Comment by caseycrogers** [comment]
Bump?

I believe the problem I'm having is related:
I'm using the integration test package and [integration_test_driver_expanded.dart](https://github.com/flutter/flutter/blob/master/packages/integration_test/lib/integration_test_driver_extended.dart) to take screenshots of my application. I'm on Windows.

if I pump once and take a screenshot, the tests run and the screenshot is produced. However, if I pump more than once before taking a screenshot, it hangs. If I pump more than once, but do not take a screenshot, it also hangs. 

I'm attempting to update the following workaround but cannot seem to adapt it as it appears `IsolateRef.load()` does not exist:
https://github.com/flutter/flutter/issues/73355#issuecomment-766961956
Has anyone else been able to revive this workaround?

Happy to open a brand new bug if it turns out my problem is distinct from this existing issue.

**Comment by Abhi011999** [comment]
@davidmartos96 like your `pumpUntilFound()` can there be something like `pumpUntilVisible()` ?

**Comment by caseycrogers** [comment]
> I was able to work around a similar issue by modifying `integrationDriver` a bit at integration_test.dart. Does it work for you?
> .
> .
> .
> ```

For those interested, I've updated this snippet to Flutter 2.0, but I have not been able to verify that it works as intended:
```dart
class IsolatesWorkaround {
  IsolatesWorkaround(this._driver, {this.log = false});

  final FlutterDriver _driver;
  final bool log;

  StreamSubscription<Event>? _streamSubscription;

  /// workaround for isolates
  /// https://github.com/flutter/flutter/issues/24703
  Future<void> resumeIsolates() async {
    final vm = await _driver.serviceClient.getVM();
    // // unpause any paused isolated
    for (final IsolateRef isolateRef in vm.isolates!) {
      final Isolate isolate =
          await _driver.serviceClient.getIsolate(isolateRef.id!);

      if (isolate.pauseEvent?.kind == 'Pause') {
        await _driver.serviceClient.resume(isolate.id!);
        if (log) {
          print('Resuming isolate: ${isolate.number}:${isolate.name}');
        }
      }
    }
    if (_streamSubscription != null) {
      return;
    }
    _streamSubscription = _driver.serviceClient.onIsolateEvent
        .asBroadcastStream()
        .listen((event) async {
          // I don't get any events when running this on my system with my app
         // so I can't verify that this section works.
          if (event.kind != 'IsolateRunnable') {
            return;
          }
          final Isolate isolate =
              await _driver.serviceClient.getIsolate(event.isolate!.id!);
          if (isolate.pauseEvent?.kind == 'Pause') {
            await _driver.serviceClient.resume(isolate.id!);
            if (log) {
              print('Resuming isolate: ${isolate.number}:${isolate.name}');
            }
          }
        });
  }

  void dispose() {
    _streamSubscription?.cancel();
  }
}
```

I turns out my issue was eliminated by removing Google banner ad during integration tests-for some reason fetching ads was making the tests hang. I don't know if this is actually related to this bug or not.

**Comment by maheshmnj** [comment]
Hi @drewjbartlett, I tried running your integration test using  [integration_test](https://docs.flutter.dev/cookbook/testing/integration/introduction) plugin on the latest master channel and the test completes properly . Closing as this is no longer reproducible in case you still encounter this issue, Please feel free to write in the comments and we will reopen it.

<details>
<summary>logs</summary>

```console
mahesh@Maheshs-MacBook-Air-M1 flutter_issue % flutter test integration_test 
Could not load custom device from config index 0: Expected enabled to be a boolean.
Multiple devices found:
sdk gphone arm64 (mobile) • emulator-5554                        • android-arm64  • Android 11 (API 30) (emulator)
iPhone 12 Pro (mobile)    • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios            •
com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
macOS (desktop)           • macos                                • darwin-arm64   • macOS 13.1 22C65 darwin-arm64
Chrome (web)              • chrome                               • web-javascript • Google Chrome 108.0.5359.124
[1]: sdk gphone arm64 (emulator-5554)
[2]: iPhone 12 Pro (026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4)
[3]: macOS (macos)
[4]: Chrome (chrome)
Please choose one (To quit, press "q/Q"): 1
00:00 +0: loading /Users/mahesh/Desktop/flutter_issue/integration_test/app_test.dart                                   R00:14 +0: loading /Users/mahesh/Desktop/flutter_issue/integration_test/app_test.dart                              14.6s
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
00:16 +0: loading /Users/mahesh/Desktop/flutter_issue/integration_test/app_test.dart                                   I00:17 +0: loading /Users/mahesh/Desktop/flutter_issue/integration_test/app_test.dart                            1,303ms
00:21 +0: it shows the home tab                                                                                        
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: exactly one matching node in the widget tree
  Actual: _KeyFinder:<zero widgets with key [<'tab-home'>] (ignoring offstage widgets)>
   Which: means none were found but one was expected

When the exception was thrown, this was the stack:
#4      main.<anonymous closure> (file:///Users/mahesh/Desktop/flutter_issue/integration_test/app_test.dart:17:5)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///Users/mahesh/Desktop/flutter_issue/integration_test/app_test.dart line 17
The test description was:
  it shows the home tab
════════════════════════════════════════════════════════════════════════════════════════════════════
00:21 +0 -1: it shows the home tab [E]                                                                                 
  Test failed. See exception logs above.
  The test description was: it shows the home tab
  

To run this test again: /Users/mahesh/Development/flutter_master/bin/cache/dart-sdk/bin/dart test /Users/mahesh/Desktop/flutter_issue/integration_test/app_test.dart -p vm --plain-name 'it shows the home tab'
00:22 +0 -1: Some tests failed.       
```
</details> 

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel master, 3.7.0-3.0.pre.33, on macOS 12.6 21G115 darwin-arm64, locale en-IN)
    • Flutter version 3.7.0-3.0.pre.33 on channel master 
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5201856805 (38 minutes ago), 2022-12-05 18:27:21 -0800
    • Engine revision a309d239c4
    • Dart version 2.19.0 (build 2.19.0-463.0.dev)
    • DevTools version 2.20.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly
      to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.55.20221129

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 108.0.5359.94

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
```
[✓] Flutter (Channel stable, 3.3.9, on macOS 12.6 21G115 darwin-arm, locale en-IN)
    • Flutter version 3.3.9 on channel stable at /Users/mahesh/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b8f7f1f986 (24 hours ago), 2022-11-23 06:43:51 +0900
    • Engine revision 8f2221fbef
    • Dart version 2.18.5
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.53.20221101

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • 026D5789-9E78-4AD5-B1B2-3F8D4E7F65E4 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>
